File: split/newdisc.pas
function newdisc: halfword;
var
  p: halfword;
begin
  p := getnode(2);
  mem[p].hh.b0 := 7;
  mem[p].hh.b1 := 0;
  mem[p + 1].hh.lh := 0;
  mem[p + 1].hh.rh := 0;
  newdisc := p;
end;

File: cpp/newdisc.cpp
#include "newdisc.h"
#include "getnode.h"

halfword newdisc(void)
{
	auto p = getnode(2);
	type(p) = disc_node;
	replace_count(p) = 0;
	pre_break(p) = 0;
	post_break(p) = 0;
	return p;
}
